common/lib/xmodule/xmodule/video_module/transcripts_utils.py

```diff
@@ -107,8 +134,19 @@ def get_transcripts_from_youtube(youtube_id, settings, i18n):
 
     youtube_text_api = copy.deepcopy(settings.YOUTUBE['TEXT_API'])
     youtube_text_api['params']['v'] = youtube_id
+    # if the transcript name is not empty on youtube server we have to pass
+    # name param in url in order to get transcript
+    # example http://video.google.com/timedtext?lang=en&v={VideoId}&name={transcript_name}
+    if youtube_transcript_name:
+        youtube_text_api['params']['name'] = youtube_transcript_name
     data = requests.get('http://' + youtube_text_api['url'], params=youtube_text_api['params'])
 
+    # Re-attempt to get transcript with name param in url from youtube server
+    if data.status_code == 200 and not data.text:
+        transcript_name_youtube_server = transcript_name_youtube_video(youtube_text_api)
+        if transcript_name_youtube_server:
+            return get_transcripts_from_youtube(youtube_id, settings, i18n, transcript_name_youtube_server)
```

Comment:
I'm confused here? Is it the case that sometimes we get the transcript without passing the `name` parameter to youtube? Or is it that we always need to pass it, and sometimes we do have it?

If it's the latter, can't you just check to see if  youtube_text_api['params']['name'] is set, and, if not, set it to `transcript_name_youtube_video(youtube_text_api)`?

If it's the former, there's no reason to do this recursively.